stumpy/floss.py

```diff
@@ -626,8 +693,14 @@ def P_(self):
     def I_(self):
         """
         Get the updated (right) matrix profile indices
+
+        The indices stored in `self.I_` reflect the starting index of
+        subsequneces with respect to the original time series.
         """
-        return self._mp[:, 3].astype(np.int64)
+        I = self._mp[:, 3].astype(np.int64).copy()
+        I[self._mp[:, 0] == np.inf] = -1
```

Comment:
I would feel more comfortable checking both condition with something like:

```
mask = np.logical_or(self._mp[:, 0] == np.inf, self._mp[:, 3] == np.arange(len(self._mp)))
I[mask] = -1
```

Reply:
I understand your concern as I think you want to be safer here. However, you may want to reconsider it as the new check, i.e `self._mp[:, 3] == np.arange(len(self._mp)`, can be only useful at the initial step where floss object is initialized. Even in that case, the main check `self._mp[:, 0] == np.inf` should be clear and less complicated. Note that with each call of method `.update`, we shift the indices to the left and add a new value `last_idx` at the end which is bigger than `len(mp)`. In other words, `elf._mp[:, 3] == np.arange(len(self._mp))` becomes useless as soon as we start using `.update()`

Reply:
> In other words, self._mp[:, 3] == np.arange(len(self._mp)) becomes useless as soon as we start using .update()

Ahh, good point! Can you please add a note here with something like:

```
# Note that we avoid comparing the right matrix profile index value with the # self index position (i.e., self._mp[:, 3] == np.arange(len(self._mp))
# because things are constantly being shifted (ingress+egress) to the left
# and so the self._mp[:, 3] == np.arange(len(self._mp)) becomes useless as # soon as we start using .update().
```

Maybe something like that?

Reply:
Sure. I will add a comment regarding this matter.